src/meta/api/src/schema_api_impl.rs

```diff
@@ -848,6 +864,224 @@ impl<KV: kvapi::KVApi<Error = MetaError>> SchemaApi for KV {
+                get_pb_value(self, &index_id_list_key).await?;
+
+            let mut index_id_list = if index_id_list_seq == 0 {
+                IndexIdList::new()
+            } else {
+                match index_id_list_opt {
+                    Some(list) => list,
+                    None => IndexIdList::new(),
+                }
+            };
+
+            // Create index by inserting these record:
+            // (tenant, index_name) -> index_id
+            // (index_id) -> index_meta
+            // append index_id into __fd_index_id_list/<tenant>/<table_id>
+            // (index_id) -> (tenant,index_name)
+
+            let index_id = fetch_id(self, IdGenerator::index_id()).await?;
+            let id_key = IndexId { index_id };
+            let id_to_name_key = IndexIdToName { index_id };
+
+            debug!(index_id, index_key = debug(&tenant_index), "new index id");
+
+            {
+                // append index_id into index_id_list
+                index_id_list.append(index_id);
+
+                let condition = vec![
+                    txn_cond_seq(tenant_index, Eq, 0),
+                    txn_cond_seq(&id_to_name_key, Eq, 0),
+                    txn_cond_seq(&index_id_list_key, Eq, index_id_list_seq),
+                ];
+                let if_then = vec![
+                    txn_op_put(tenant_index, serialize_u64(index_id)?), /* (tenant, index_name) -> index_id */
+                    txn_op_put(&id_key, serialize_struct(&req.meta)?),  // (index_id) -> index_meta
+                    txn_op_put(&index_id_list_key, serialize_struct(&index_id_list)?), /* _fd_index_id_list/<tenant> -> index_id_list */
+                    txn_op_put(&id_to_name_key, serialize_struct(tenant_index)?), /* __fd_index_id_to_name/<index_id> -> (tenant,index_name) */
+                ];
+
+                let txn_req = TxnRequest {
+                    condition,
+                    if_then,
+                    else_then: vec![],
+                };
+
+                let (succ, _responses) = send_txn(self, txn_req).await?;
+
+                debug!(
+                    index_name = debug(&tenant_index),
+                    id = debug(&id_key),
+                    succ = display(succ),
+                    "create_index"
+                );
+
+                if succ {
+                    return Ok(CreateIndexReply { index_id });
+                }
+            }
+        }
+
+        Err(KVAppError::AppError(AppError::TxnRetryMaxTimes(
+            TxnRetryMaxTimes::new("create_index", TXN_MAX_RETRY_TIMES),
+        )))
+    }
+
+    #[tracing::instrument(level = "debug", ret, err, skip_all)]
+    async fn drop_index(&self, req: DropIndexReq) -> Result<DropIndexReply, KVAppError> {
+        debug!(req = debug(&req), "SchemaApi: {}", func_name!());
+
+        let tenant_index = &req.name_ident;
+        let mut retry = 0;
+
+        while retry < TXN_MAX_RETRY_TIMES {
+            retry += 1;
```

Comment:
This will make txn retry simpler:
https://github.com/datafuselabs/databend/blob/c3386ac22dc11aab1bf99d904a1666716bbeda57/src/meta/api/src/schema_api_impl.rs#L1980-L1982